lib/n_plus_one_control/rspec/matchers/perform_linear_number_of_queries.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+# rubocop:disable  Metrics/BlockLength
+::RSpec::Matchers.define :perform_linear_number_of_queries do |slope: 1, intercept: 1|
```

Comment:
What do you think about simplifying thing a bit and removing the intercept? I think, knowing the exact formula for queries(N) is too much.

Instead, we can calculate the actual slope and compare it with the expected one.

Smth like:

```ruby
# @queries[i][0] is a scale factor, @queries[i][1] contains the list of queries
actual_slope = (@queries[0][1].size - @queries[1][1].size) / (@queries[0][0] - @queries[1][0])

actual_slope <= slope
```

Reply:
I agree that  knowing the intercept is too much and it's not easy to know the exact intercept in practice.